service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -269,6 +269,25 @@ public List<Workspace> getWorkspacesMatchingList(List<UUID> idList, int offset,
     return jdbcTemplate.query(sql, params, WORKSPACE_ROW_MAPPER);
   }
 
+  @ReadTransaction
+  public UUID getUuidByUserFacingId(String userFacingId) {
```

Comment:
This could retrieve the workspace directly in one transaction instead of looping out and back with the UUID to look up the same row again.

Reply:
Done, makes sense